client/src/components/Dashboard.js

```diff
@@ -6,33 +6,117 @@ import TableContainer from '@material-ui/core/TableContainer';
 import TableHead from '@material-ui/core/TableHead';
 import TableRow from '@material-ui/core/TableRow';
 import Paper from '@material-ui/core/Paper';
+import TablePagination from '@material-ui/core/TablePagination';
+import TableFooter from '@material-ui/core/TableFooter';
 import Row from '../components/Row';
+import Modal from '@material-ui/core/Modal';
+import History from './History';
+import { makeStyles } from '@material-ui/core/styles';
 
-export default function Dasboard(props) {
-    const { rows } = props;
-    if (rows.length > 0) {
-        debugger;
+class Dashboard extends React.Component {
+// (props) {
+    constructor(props) {
+        super(props);
+        this.state = {
+            page: 0,
+            rowsPerPage: 5,
+            open: false,
+            history: []
+        };
+        this.handleModalOpen = this.handleModalOpen.bind(this);
+        this.handleHistory = this.handleHistory.bind(this);
     }
+    // const [page, setPage] = React.useState(0);
+    // const [rowsPerPage, setRowsPerPage] = React.useState(5);
+    // const [open, setOpen] = React.useState(false);
+    // const [history, setHistory] = React.useState();
+    // const { rows } = props;
+    // if (this.props.rows.length > 0) {
+    //     debugger;
+    // }
 
-    return (
-        <TableContainer component={Paper}>
-            <Table aria-label="collapsible table">
-                <TableHead>
-                <TableRow>
-                    <TableCell />
-                    <TableCell><b>Reclamante/Requerente</b></TableCell>
-                    <TableCell><b>Reclamadas</b></TableCell>
-                    <TableCell><b>ADV</b></TableCell>
-                    <TableCell><b>Vara</b></TableCell>
-                    <TableCell><b>Processo</b></TableCell>
-                </TableRow>
-                </TableHead>
-                <TableBody>
-                {rows.map((row) => (
-                    <Row key={row.name} row={row} />
-                ))}
-                </TableBody>
-            </Table>
-        </TableContainer>
-    );
+    handleModalOpen = () => {
+        this.setState({open: !this.state.open});
+    };
+
+    handleHistory = (value) => {
+        console.log(value)
+        this.setState({history: value});
+        console.log(this.state)
+    };
+
+    handleChangePage = (event, newPage) => {
+        this.setState({page: newPage});
+    };
+    
+    handleChangeRowsPerPage = (event) => {
+        this.setState({rowsPerPage: parseInt(event.target.value, 10)});
+        this.setState({page: 0});
+    };
+
+    useStyles = makeStyles((theme) => ({
+        paper: {
+            position: 'absolute',
+            width: 400,
+            backgroundColor: theme.palette.background.paper,
+            border: '2px solid #000',
+            boxShadow: theme.shadows[5],
+            padding: theme.spacing(2, 4, 3),
+            },
+    }));
+
+    render(){
+        return (
+            <>
```

Comment:
Não consigo compilar aqui. Dá erro nessa linha.

Trocar por div?

Reply:
sabe dizer qual erro que dá?

Reply:
Expression needs a parent element

Ou algo assim. Daqui a pouco vejo direitinho e coloco aqui

Reply:
Dá "Syntax error: Unexpected token". Se remover esse `<>`  dá "Syntax error: Adjacent JSX elements must be wrapped in an enclosing tag"

Eu troquei o `<>` por um `<div>` e compilou.

Reply:
```suggestion
            <div>
```